pkg/spec/resources_transform.go

```diff
@@ -130,49 +128,6 @@ func createServices(app *App) ([]runtime.Object, error) {
 	return svcs, nil
 }
 
-// Creates a Deployment Kubernetes resource. The returned Deployment resource
-// will be nil if it could not be generated due to insufficient input data.
-func createDeployment(app *App) (*ext_v1beta1.Deployment, error) {
-
-	// We need to error out if both, app.PodSpec and app.DeploymentSpec are empty
-	if reflect.DeepEqual(app.PodSpec, api_v1.PodSpec{}) && reflect.DeepEqual(app.DeploymentSpec, ext_v1beta1.DeploymentSpec{}) {
-		log.Debug("Both, app.PodSpec and app.DeploymentSpec are empty, not enough data to create a deployment.")
-		return nil, nil
-	}
-
-	// We are merging whole DeploymentSpec with PodSpec.
-	// This means that someone could specify containers in template.spec and also in top level PodSpec.
-	// This stupid check is supposed to make sure that only one of them set.
-	// TODO: merge DeploymentSpec.Template.Spec and top level PodSpec
-	if !(reflect.DeepEqual(app.DeploymentSpec.Template.Spec, api_v1.PodSpec{}) || reflect.DeepEqual(app.PodSpec, api_v1.PodSpec{})) {
-		return nil, fmt.Errorf("Pod can't be specfied in two places. Use top level PodSpec or template.spec (DeploymentSpec.Template.Spec) not both")
-	}
-
-	deploymentSpec := app.DeploymentSpec
-
-	// top level PodSpec is not empty, use it for deployment template
-	// we already know that if app.PodSpec is not empty app.DeploymentSpec.Template.Spec is empty
-	if !reflect.DeepEqual(app.PodSpec, api_v1.PodSpec{}) {
-		deploymentSpec.Template.Spec = app.PodSpec
-	}
-
-	// TODO: check if this wasn't set by user, in that case we shouldn't ovewrite it
-	deploymentSpec.Template.ObjectMeta.Name = app.Name
-
-	// TODO: merge with already existing labels and avoid duplication
-	deploymentSpec.Template.ObjectMeta.Labels = app.Labels
-
-	deployment := ext_v1beta1.Deployment{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   app.Name,
-			Labels: app.Labels,
-		},
-		Spec: deploymentSpec,
-	}
-
-	return &deployment, nil
-}
-
 // create PVC reading the root level persistent volume field
 func createPVC(v VolumeClaim, labels map[string]string) (*api_v1.PersistentVolumeClaim, error) {
```

Comment:
All other functions are in the form of `func (app *ControllerFields) createSomething ....`
Why we don't have `func (app *ControllerFields) createPVC(labels map[string]string) (*api_v1.PersistentVolumeClaim, error)`?

Reply:
we can even remove `labels` from parameters, as you can get them from `app` struct in the same way as it's done in the other functions

Reply:
fixed